gatling-maven-plugin/src/main/java/io/gatling/mojo/GatlingMojo.java

```diff
@@ -204,8 +212,20 @@ public void execute() throws MojoExecutionException {
 
         } else {
           Collection<String> simulations = simulations();
+          boolean buildFails = false;
           for (String simulation : simulations) {
-            executeGatling(jvmArgs, gatlingArgs(simulation), testClasspath, toolchain);
+            try {
+              executeGatling(jvmArgs, gatlingArgs(simulation), testClasspath, toolchain);
+            } catch (Exception e) {
+              if (!continueAfterFail) {
+                throw new GatlingSimulationAssertionsFailedException(e);
+              }
+              buildFails = true;
+              getLog().warn("There was some errors while running your simulation, but continueAfterFail set to true, so your simulations continue to perform.");
+            }
+          }
+          if (buildFails) {
+            throw new MojoExecutionException("Gatling failed");
```

Comment:
No. I think you must re-throw the first GatlingSimulationAssertionsFailedException.